Engines/Wine/Verbs/D9VK/script.js

```diff
@@ -12,17 +12,22 @@ include("utils.functions.filesystem.files");
  */
 Wine.prototype.D9VK = function (d9vkVersion) {
     var operatingSystemFetcher = Bean("operatingSystemFetcher");
-
+    var uiQuestionFactory = Bean("uiQuestionFactory");
     print("NOTE: Wine version should be greater or equal to 3.10");
-
-    if (operatingSystemFetcher.fetchCurrentOperationSystem() != "Linux") {
-        this.wizard().message(tr("D9VK might not work correctly on macOS. This is depending on Metal api support and MoltenVK compatibility layer advancement"));
+    if (operatingSystemFetcher.fetchCurrentOperationSystem().getFullName() !== "Linux")
+    {
+        uiQuestionFactory.create(
+				tr("D9VK is currently unsupported on non-Linux operating systems due to MoltenVK implementation being incomplete. Do you want to continue? Choosing yes will skip DXVK verb installation and continue with other verbs. Choosing no will quit script installation."),
+				() => return this;
```

Comment:
This looks like a syntax error to me:
1) when using the arrow syntax without `{...}` I don't think you are allowed to use `;` at the end
2) I don't think that this `return` allows you to exit the outer method (i.e. `Wine.D9VK(d9vkVersion)`), because it is part of an inner function

Have you tested this call to be sure it that it works as it should?

Reply:
No because I don't have a mac so the statement in that if is always false.

Reply:
The example assigns the lambda function (i.e. `([a, b] = [1, 2], {x: c} = {x: a + b}) => a + b + c;`) to a variable (i.e. `var f`). This assignment is terminated with the `;` and not the lambda expression itself.

Have you tried executing `eslint -c .eslintrc.yml --fix ./**/*.js` to see if it shows any issues with the script?

Reply:
I get this:
```
  16:11  error  Parsing error: Unexpected token return
```

Reply:
This means that there is a syntax error, because `return` doesn't belong there.

Reply:
To make this work we will need to add an additional method to the `UiQuestionFactory`. This method needs to be blocking. I think it should have the following signature:

```java
/**
 * Asks the user a question where he/she can answer either yes or no
 * @param questionText The question text
 * @returns True if the user answered yes and false if he answered no
 */
boolean create(String questionText);
```

Reply:
@Zemogiter I've just added a new PR for phoenicis that implements the above suggested method.

Can you please try if it works as intended?
I think you will need something like:

```javascript
const answer = uiQuestionFactory.create(
   tr("D9VK is currently unsupported on non-Linux operating systems due to MoltenVK implementation being incomplete. Do you want to continue? Choosing yes will skip DXVK verb installation and continue with other verbs. Choosing no will quit script installation.")
);

// or: `if (answer == false)` 
if (!answer) {
   return this;
}
```

Reply:
I still think there should be three possible answer:
1. Install D9VK anyways (to test MoltenVK)
2. Skip D9VK installation
3. Stop whole app script.

Reply:
Thank you. After this change the error is gone and the D9VK installation goes fine.

Reply:
Could you add then in the displayed message that, if they choose to continue, they can still install the verb from the container tab if the want to test it ?

Reply:
Can't we support more complex questions via (taken from the League of Legends script):

```
var regions = [
            "EU West",
            "Latin America North",
            "Latin America South",
            "Oceania",
            "Japan",
            "Turkey",
            "Brasil",
            "EU Nordic & East",
            "North America",
            "Russia"
        ];
var selectedRegion = wizard.menu(tr("Select your region:"), regions);
```

Reply:
No because no matter whtever the script is executed from other script or via verbs tab they will always get that message.

Reply:
Ok, so we basically we need to do this via a popup dialog?

We could also think about adding another method to `uiQuestionFactory`:

```java
String create(String questionText, List<String> choices, String defaultChoice);
```

This method takes a question text and a list with choices. After the user has selected one of the choices the choice text is returned.

Would such a method help?

Reply:
@madoar Yeah that sounds more clear than a single yes/no question since we have three options (install D9VK anyways, skip D9VK and go with other verbs and the rest of the script, quit the script alltogether)

Reply:
@Zemogiter I've just added a new `MultipleChoiceDialog` which can be opened with the previously mentioned command. Can you give it a try?

Please be aware that I didn't test the new dialog. So if it looks kind of strange don't hesitate to tell me so I can fix it (ideally with a screenshot)

Reply:
I'm confused as to how the `MultipleChoiceDialog` call should look like.

Reply:
```javascript
const answer = uiQuestionFactory.create(
   tr("D9VK is currently unsupported on non-Linux operating systems due to MoltenVK implementation being incomplete. Do you want to continue? Choosing yes will skip DXVK verb installation and continue with other verbs. Choosing no will quit script installation."),
   ["Choice 1", "Choice 2", "Exit Installer"]
);

if (!answer || answer == "Exit Installer") {
   return this;
}
```

Reply:
it might also be interesting to bypass d9vk installation on computer which does not have enough vulkan support, because wine's d3d9 to OpenGL can to the job.

Reply:
Yeah but how we determine vulkan support? Looking at what gpu the user have?

Reply:
Just ask the user if he is sure to install it. I am currently working on a way get such infos.

Reply:
OK I'll wait until that PR is merged into POL5.